src/uct/ib/mlx5/ib_mlx5_dv.c

```diff
@@ -235,8 +238,8 @@ static ucs_status_t uct_ib_mlx5dv_md_open(struct ibv_device *ibv_device,
                                                    (UCT_IB_MLX5_CAP_GENERAL << 1));
     ret = mlx5dv_devx_general_cmd(ctx, in, sizeof(in), out, sizeof(out));
     if (ret == 0) {
-        if (!UCT_IB_MLX5DV_GET(cmd_hca_cap, cap, dct)) {
-            has_dc = 0;
+        if (UCT_IB_MLX5DV_GET(cmd_hca_cap, cap, dct)) {
+            dev->flags |= UCT_IB_DEVICE_FLAG_DC;
```

Comment:
i would also move UCT_IB_DEVICE_FLAG_DC to uct_ib_mlx5_md (maybe not in this pr)

Reply:
Would be nice to combine flags refactoring with uct_ib_device_spec_t removal